language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/builder/ResourcePathCompletionUtil.java

```diff
@@ -0,0 +1,362 @@
+        CompletionItem item = build(functionSymbol, functionSignature, context);
+        item.setFilterText(getFilterTextForResourceMethod(functionSymbol));
+        return item;
+    }
+
+    /**
+     * Creates and returns a completion item.
+     *
+     * @param resourceMethodSymbol resource method symbol.
+     * @param segments             path segments.
+     * @param context              LS context
+     * @return {@link CompletionItem}
+     */
+    public static CompletionItem build(ResourceMethodSymbol resourceMethodSymbol,
+                                       List<PathSegment> segments,
+                                       BallerinaCompletionContext context) {
+        Pair<String, String> functionSignature = ResourcePathCompletionUtil
+                .getResourceAccessSignature(resourceMethodSymbol, context, segments);
+        CompletionItem item = build(resourceMethodSymbol, functionSignature, context);
+        item.setFilterText(getFilterTextForResourceMethod(resourceMethodSymbol, segments));
+        return item;
+    }
+
+    private static CompletionItem build(ResourceMethodSymbol resourceMethodSymbol,
+                                        Pair<String, String> functionSignature,
+                                        BallerinaCompletionContext context) {
+
+        CompletionItem item = new CompletionItem();
+        FunctionCompletionItemBuilder.setMeta(item, resourceMethodSymbol, context);
+        item.setLabel(functionSignature.getRight());
+        item.setInsertText(functionSignature.getLeft());
+
+        //Add additional text edits
+        checkAndSetAdditionalTextEdits(item, context);
+        return item;
+    }
+
+    /**
+     * Check if the cursor is positioned with in the method call context of client resource access action node.
+     *
+     * @param node    client resource access action node.
+     * @param context completion context
+     * @return boolean
+     */
+    public static boolean isInMethodCallContext(ClientResourceAccessActionNode node,
+                                                BallerinaCompletionContext context) {
+        return node.dotToken().isPresent()
+                && node.dotToken().get().textRange().endOffset() <= context.getCursorPositionInTree();
+    }
+
+    private static void checkAndSetAdditionalTextEdits(CompletionItem item,
+                                                       BallerinaCompletionContext context) {
+        //Check and replace preceding slash token and dot
+        Token token = null;
+        Optional<ClientResourceAccessActionNode> node = findClientResourceAccessActionNode(context);
+        if (node.isPresent()) {
+            if (isInMethodCallContext(node.get(), context)) {
+                if (node.get().resourceAccessPath().isEmpty()
+                        && !node.get().slashToken().isMissing()
+                        && node.get().dotToken().isPresent()) {
```

Comment:
I think we don't need this dot token is present check since it's done in `isInMethodCallContext()` method